sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoinExec.scala

```diff
@@ -578,7 +594,9 @@ case class ShuffledHashJoinExec(
          |
          |  if (!$foundMatch) {
          |    $buildRow = null;
-         |    $consumeFullOuterJoinRow();
+         |    if ($isFullOuterJoin) {
+         |      $consumeFullOuterJoinRow();
+         |    }
```

Comment:
Seems `isFullOuterJoin` could be pull to outside `if`?

```Scala
if (!$foundMatch && $isFullOuterJoin) {
  $buildRow = null;
  $consumeFullOuterJoinRow();
}
```

Reply:
I was afraid of side effect if I don't set buildRow to null, if not fullOuterJoin.  Let me check.

Reply:
I end up pulling this in separate if block, leaving to set buildRow=null in both case if not found match.  I was thinking its cleaner, to clear the state, and more in line with uniqueKey side.